### ISSUE TITLE
Frost 1D model output for  time series looks flawed

### ISSUE THREAD

**Issue by iovereem** [permamodel/permamodel#39]
I tested the Frost Air 1D model by using two timeseries: Barrow 1961-2015 warmest month and coldest month. In WMT.
The 55 years simulation succesfully runs, but the output for the Frost number looks flawed. it reports a few varying values for the first few years, then reports back Frost number = 0.222 for the rest of the simulation. 

Input files are attached. 

[T-highmonthBarrow.txt](https://github.com/permamodel/permamodel/files/985050/T-highmonthBarrow.txt)
[Tlowmonth_Barrow.txt](https://github.com/permamodel/permamodel/files/985051/Tlowmonth_Barrow.txt)

Output file format is not good for Github: so here is the text array 

Year: 1961  output=('0.633', '-1.000', '-1.000')
Year: 1962  output=('0.500', '-1.000', '-1.000')
Year: 1963  output=('0.367', '-1.000', '-1.000')
Year: 1964  output=('0.222', '-1.000', '-1.000')
Year: 1965  output=('0.222', '-1.000', '-1.000')
Year: 1966  output=('0.222', '-1.000', '-1.000')
Year: 1967  output=('0.222', '-1.000', '-1.000')
Year: 1968  output=('0.222', '-1.000', '-1.000')
Year: 1969  output=('0.222', '-1.000', '-1.000')
Year: 1970  output=('0.222', '-1.000', '-1.000')
Year: 1971  output=('0.222', '-1.000', '-1.000')
Year: 1972  output=('0.222', '-1.000', '-1.000')
Year: 1973  output=('0.222', '-1.000', '-1.000')
Year: 1974  output=('0.222', '-1.000', '-1.000')
Year: 1975  output=('0.222', '-1.000', '-1.000')
Year: 1976  output=('0.222', '-1.000', '-1.000')
Year: 1977  output=('0.222', '-1.000', '-1.000')
Year: 1978  output=('0.222', '-1.000', '-1.000')
Year: 1979  output=('0.222', '-1.000', '-1.000')
Year: 1980  output=('0.222', '-1.000', '-1.000')
Year: 1981  output=('0.222', '-1.000', '-1.000')
Year: 1982  output=('0.222', '-1.000', '-1.000')
Year: 1983  output=('0.222', '-1.000', '-1.000')
Year: 1984  output=('0.222', '-1.000', '-1.000')
Year: 1985  output=('0.222', '-1.000', '-1.000')
Year: 1986  output=('0.222', '-1.000', '-1.000')
Year: 1987  output=('0.222', '-1.000', '-1.000')
Year: 1988  output=('0.222', '-1.000', '-1.000')
Year: 1989  output=('0.222', '-1.000', '-1.000')
Year: 1990  output=('0.222', '-1.000', '-1.000')
Year: 1991  output=('0.222', '-1.000', '-1.000')
Year: 1992  output=('0.222', '-1.000', '-1.000')
Year: 1993  output=('0.222', '-1.000', '-1.000')
Year: 1994  output=('0.222', '-1.000', '-1.000')
Year: 1995  output=('0.222', '-1.000', '-1.000')
Year: 1996  output=('0.222', '-1.000', '-1.000')
Year: 1997  output=('0.222', '-1.000', '-1.000')
Year: 1998  output=('0.222', '-1.000', '-1.000')
Year: 1999  output=('0.222', '-1.000', '-1.000')
Year: 2000  output=('0.222', '-1.000', '-1.000')
Year: 2001  output=('0.222', '-1.000', '-1.000')
Year: 2002  output=('0.222', '-1.000', '-1.000')
Year: 2003  output=('0.222', '-1.000', '-1.000')
Year: 2004  output=('0.222', '-1.000', '-1.000')
Year: 2005  output=('0.222', '-1.000', '-1.000')
Year: 2006  output=('0.222', '-1.000', '-1.000')
Year: 2007  output=('0.222', '-1.000', '-1.000')
Year: 2008  output=('0.222', '-1.000', '-1.000')
Year: 2009  output=('0.222', '-1.000', '-1.000')
Year: 2010  output=('0.222', '-1.000', '-1.000')
Year: 2011  output=('0.222', '-1.000', '-1.000')
Year: 2012  output=('0.222', '-1.000', '-1.000')
Year: 2013  output=('0.222', '-1.000', '-1.000')
Year: 2014  output=('0.222', '-1.000', '-1.000')
Year: 2015  output=('0.222', '-1.000', '-1.000')

**Comment by sc0tts** [comment]
This is good info to have.  I don't remember if the original Frost Number
(1D) was built to handle a time series, but I should be able add that
capability.

Also, the extra "-1.000"s in each year's outputs are for "surface frost
number" and "Stefan frost number" respectively.  Shall I remove them from
the output when they aren't calculated in the code?  That would be
cleaner....

-Scott

I tested the Frost Air 1D model by using two timeseries: Barrow 1961-2015

**Comment by mdpiper** [comment]
The extra FN surface and Stefan outputs aren't important right now; they're not used by WMT.

We do need to figure out why FNM is only giving four output values, though. I get the same result as @iovereem if I step through FNM in PyMT. I believe FNM isn't reading from the file -- the output values are the same as if you run from defaults.

**Comment by iovereem** [comment]
The original Frost Number model may not have timeseries, but the functionality in WMT easily allows it, so if it is a trivial fix, it seems to gain much added functionality.  

I agree with Mark on the extra -1.000, we can leave that for now. The WMT netcdf output only has the Air Frost number output.  But github does not allow uploads to this issues tabs of netcdf files, why I opened the *.dat file instead.

**Comment by sc0tts** [comment]
This should be fixed with Pull Request #43

**Comment by mdpiper** [comment]
Fixed. Try the public model "FNM Barrow series" in the **wmt-permafrost** instance.

Here's the output:

```
Year: 1961  output=('0.810', '-1.000', '-1.000')
Year: 1962  output=('0.791', '-1.000', '-1.000')
Year: 1963  output=('0.829', '-1.000', '-1.000')
Year: 1964  output=('0.859', '-1.000', '-1.000')
Year: 1965  output=('0.799', '-1.000', '-1.000')
Year: 1966  output=('0.837', '-1.000', '-1.000')
Year: 1967  output=('0.847', '-1.000', '-1.000')
Year: 1968  output=('0.771', '-1.000', '-1.000')
Year: 1969  output=('0.879', '-1.000', '-1.000')
Year: 1970  output=('0.898', '-1.000', '-1.000')
Year: 1971  output=('0.811', '-1.000', '-1.000')
Year: 1972  output=('0.790', '-1.000', '-1.000')
Year: 1973  output=('0.784', '-1.000', '-1.000')
Year: 1974  output=('0.819', '-1.000', '-1.000')
Year: 1975  output=('0.860', '-1.000', '-1.000')
Year: 1976  output=('0.833', '-1.000', '-1.000')
Year: 1977  output=('0.768', '-1.000', '-1.000')
Year: 1978  output=('0.840', '-1.000', '-1.000')
Year: 1979  output=('0.738', '-1.000', '-1.000')
Year: 1980  output=('0.902', '-1.000', '-1.000')
Year: 1981  output=('0.818', '-1.000', '-1.000')
Year: 1982  output=('0.910', '-1.000', '-1.000')
Year: 1983  output=('0.844', '-1.000', '-1.000')
Year: 1984  output=('0.779', '-1.000', '-1.000')
Year: 1985  output=('0.827', '-1.000', '-1.000')
Year: 1986  output=('0.794', '-1.000', '-1.000')
Year: 1987  output=('0.819', '-1.000', '-1.000')
Year: 1988  output=('0.902', '-1.000', '-1.000')
Year: 1989  output=('0.714', '-1.000', '-1.000')
Year: 1990  output=('0.782', '-1.000', '-1.000')
Year: 1991  output=('0.844', '-1.000', '-1.000')
Year: 1992  output=('0.818', '-1.000', '-1.000')
Year: 1993  output=('0.763', '-1.000', '-1.000')
Year: 1994  output=('0.816', '-1.000', '-1.000')
Year: 1995  output=('0.789', '-1.000', '-1.000')
Year: 1996  output=('0.777', '-1.000', '-1.000')
Year: 1997  output=('0.774', '-1.000', '-1.000')
Year: 1998  output=('0.706', '-1.000', '-1.000')
Year: 1999  output=('0.796', '-1.000', '-1.000')
Year: 2000  output=('0.838', '-1.000', '-1.000')
Year: 2001  output=('0.842', '-1.000', '-1.000')
Year: 2002  output=('0.768', '-1.000', '-1.000')
Year: 2003  output=('0.782', '-1.000', '-1.000')
Year: 2004  output=('0.737', '-1.000', '-1.000')
Year: 2005  output=('0.781', '-1.000', '-1.000')
Year: 2006  output=('0.772', '-1.000', '-1.000')
Year: 2007  output=('0.726', '-1.000', '-1.000')
Year: 2008  output=('0.779', '-1.000', '-1.000')
Year: 2009  output=('0.758', '-1.000', '-1.000')
Year: 2010  output=('0.746', '-1.000', '-1.000')
Year: 2011  output=('0.792', '-1.000', '-1.000')
Year: 2012  output=('0.732', '-1.000', '-1.000')
Year: 2013  output=('0.730', '-1.000', '-1.000')
Year: 2014  output=('0.825', '-1.000', '-1.000')
Year: 2015  output=('0.792', '-1.000', '-1.000')
```
